pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -194,6 +194,19 @@ def _resolve_openai_image_generation_size(
     return mapped_size
 
 
+def _check_azure_content_filter(e: APIStatusError, system: str) -> dict[str, Any] | None:
```

Comment:
Would it be possible to do this using Pydantic models as we do here?

https://github.com/pydantic/pydantic-ai/blob/c1213e4a7a978099b64c642b38e1e041d18afad6/pydantic_ai_slim/pydantic_ai/models/groq.py#L663-L688